dart/collision/CollisionGroup.h

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2016, Georgia Tech Research Corporation
+ * All rights reserved.
+ *
+ * Author(s): Jeongseok Lee <jslee02@gmail.com>
+ *
+ * Georgia Tech Graphics Lab and Humanoid Robotics Lab
+ *
+ * Directed by Prof. C. Karen Liu and Prof. Mike Stilman
+ * <karenliu@cc.gatech.edu> <mstilman@cc.gatech.edu>
+ *
+ * This file is provided under the following "BSD-style" License:
+ *   Redistribution and use in source and binary forms, with or
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef DART_COLLISION_COLLISIONGROUP_H_
+#define DART_COLLISION_COLLISIONGROUP_H_
+
+#include <vector>
+#include "dart/collision/SmartPointer.h"
+#include "dart/collision/Option.h"
+#include "dart/collision/Result.h"
+#include "dart/dynamics/SmartPointer.h"
+
+namespace dart {
+namespace collision {
+
+class CollisionGroup
+{
+public:
+
+  /// Constructor
+  CollisionGroup(const CollisionDetectorPtr& collisionDetector);
+
+  /// Destructor
+  virtual ~CollisionGroup();
+
+  /// Return collision detection engine associated with this CollisionGroup
+  CollisionDetector* getCollisionDetector() const;
+
+  /// Register a ShapeFrame to this CollisionGroup
+  void registerShapeFrame(const dynamics::ShapeFrame* shapeFrame);
+
+  /// Register ShapeFrames to this CollisionGroup
+  void registerShapeFrames(
+      const std::vector<const dynamics::ShapeFrame*>& shapeFrames);
+
+  /// Register all the ShapeFrames in the other CollisionGroups to this
+  /// CollisionGroup
+  template <typename First, typename... Others>
+  void registerShapeFrames(const First* first, const Others*... others);
```

Comment:
Either the name or the comment of this method is misleading:

- the name suggests that it takes a variable number of `ShapeFrame *`s
- the comment suggests that it takes a variable number of `CollisionGroup *`s

At a higher level, I am in favor of moving this method entirely. I believe we can achieve the same functionality by calling the `std::vector` overload:
```c++
registerShapeFrames({shapeFrame1, shapeFrame2, ...});
```